Mjml.Net/Components/Body/SectionComponent.cs

```diff
@@ -1,11 +1,595 @@
+                    renderer.StartConditionalTag();
+                    renderer.ElementStart("td")
+                        .Attr("align", child.Node.GetAttribute("align"))
+                        .Attr("class", child.Node.GetAttribute("css-class")?.SuffixCssClasses("outlook"))
+                        .Style("vertical-align", child.Node.GetAttribute("vertical-align"))
+                        .Style("width", child.Node.GetAttribute("width"));
+                    renderer.EndConditionalTag();
+
+                    child.Render(renderer, context);
+
+                    renderer.StartConditionalTag();
+                    renderer.ElementEnd("td");
+                    renderer.EndConditionalTag();
+                }
+            }
+
+            renderer.StartConditionalTag();
+            renderer.ElementEnd("tr");
+            renderer.EndConditionalTag();
+        }
+
+        private string? GetBackground()
+        {
+            if (!HasBackground())
+            {
+                return BackgroundColor;
+            }
+
+            return $"{BackgroundColor} url({BackgroundUrl}) {GetBackgroundPositionString()} / {BackgroundSize} {BackgroundRepeat}";
+        }
+
+        private string? GetBackgroundPositionString()
+        {
+            (string parsedX, string parsedY) = ParseBackgroundPosition();
+
+            var x = string.IsNullOrEmpty(BackgroundPositionX) ? parsedX : BackgroundPositionX;
+            var y = string.IsNullOrEmpty(BackgroundPositionY) ? parsedY : BackgroundPositionY;
+
+            return $"{x} {y}";
+        }
+
+        private (string x, string y) ParseBackgroundPosition()
+        {
+            var positions = BackgroundPosition.Split(' ');
+
+            bool IsTopOrBottom(ref string axis)
+            {
+                if (axis.Equals("top", StringComparison.OrdinalIgnoreCase) || axis.Equals("bottom", StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+
+                return false;
+            }
+
+            switch (positions.Length)
+            {
+                case 1:
+                    if (IsTopOrBottom(ref positions[0]))
+                    {
+                        return ("center", positions[0]);
+                    }
+
+                    return (positions[0], "center");
+
+                case 2:
+                    if (IsTopOrBottom(ref positions[0]) || (positions[0].Equals("center", StringComparison.OrdinalIgnoreCase) && IsTopOrBottom(ref positions[1])))
+                    {
+                        return (positions[1], positions[0]);
+                    }
+
+                    return (positions[0], positions[1]);
+
+                default:
+                    return ("center", "top");
+            }
+        }
+
+        private (string xPercent, string yPercent) GetBackgroundPositionAsPercentage(ref string backgroundPositionX, ref string backgroundPositionY)
```

Comment:
Can we get rid of the "ref"? It is hard to reason about and it is a class anyway.